### PR TITLE
Ped & fixes, including peds dive into danger fix

### PR DESCRIPTION
**Pull request by erorcun**

### CODE REVIEW

File: src/peds/Ped.cpp [hunk 1]
```diff
@@ -4110,6 +4121,623 @@ CPed::SetEvasiveStep(CEntity *reason, uint8 animType)
...
+		else
+			angleToFace = vehDirection - 0.5 * PI;
+#endif
+
+		neededTurn = Abs(angleToFace - m_fRotationCur);
+
+		if (neededTurn > PI)
+			neededTurn = 2 * PI - neededTurn;
+
+		if (neededTurn <= 0.5*PI) {
```
Comment (ShFil119) [inline src/peds/Ped.cpp]: 0.5f

File: src/peds/Ped.cpp [hunk 2]
```diff
@@ -4110,6 +4121,623 @@ CPed::SetEvasiveStep(CEntity *reason, uint8 animType)
...
+
+		// We don't want to dive towards car's direction
+		float dangerZone = angleToFace - vehDirection;
+		dangerZone = CGeneral::LimitRadianAngle(dangerZone);
+
+		// So, add or subtract 90deg (jump to left/right) according to that
+		if (dangerZone > 0.0)
+			angleToFace = 0.5 * PI + vehDirection;
+		else
+			angleToFace = vehDirection - 0.5 * PI;
```
Comment (ShFil119) [inline src/peds/Ped.cpp]: 0.5f

File: src/peds/Ped.cpp [hunk 3]
```diff
@@ -4110,6 +4121,623 @@ CPed::SetEvasiveStep(CEntity *reason, uint8 animType)
...
+		if (angleToFace > PI)
+			angleToFace -= 2 * PI;
+
+		// We don't want to dive towards car's direction
+		float dangerZone = angleToFace - vehDirection;
+		dangerZone = CGeneral::LimitRadianAngle(dangerZone);
+
+		// So, add or subtract 90deg (jump to left/right) according to that
+		if (dangerZone > 0.0)
+			angleToFace = 0.5 * PI + vehDirection;
```
Comment (ShFil119) [inline src/peds/Ped.cpp]: 0.5f

File: src/peds/Ped.cpp [hunk 4]
```diff
@@ -4110,6 +4121,623 @@ CPed::SetEvasiveStep(CEntity *reason, uint8 animType)
...
+
+		if (angleToFace > PI)
+			angleToFace -= 2 * PI;
+
+		// We don't want to dive towards car's direction
+		float dangerZone = angleToFace - vehDirection;
+		dangerZone = CGeneral::LimitRadianAngle(dangerZone);
+
+		// So, add or subtract 90deg (jump to left/right) according to that
+		if (dangerZone > 0.0)
```
Comment (ShFil119) [inline src/peds/Ped.cpp]: 0.0f

File: src/peds/Ped.cpp [hunk 5]
```diff
@@ -4110,6 +4121,623 @@ CPed::SetEvasiveStep(CEntity *reason, uint8 animType)
...
+
+	if (onlyRandomJump) {
+		if (reason) {
+			// Simple version of my bug fix below. Doesn't calculate "danger zone", selects jump direction randomly.
+			// Also doesn't include random hands up, sound etc. Only used on player ped and peds running from gun shots.
+
+			float vehDirection = CGeneral::GetRadianAngleBetweenPoints(
+				veh->m_vecMoveSpeed.x, veh->m_vecMoveSpeed.y,
+				0.0f, 0.0f);
+			angleToFace = (CGeneral::GetRandomNumber() & 1) * PI + (-0.5*PI) + vehDirection;
```
Comment (ShFil119) [inline src/peds/Ped.cpp]: 0.5f

File: src/peds/Ped.cpp [hunk 6]
```diff
@@ -4067,22 +4074,26 @@ CPed::SetEvasiveStep(CEntity *reason, uint8 animType)
...
-			if (neededTurn <= 0.0)
+			if (angleToFace > PI)
+				angleToFace -=  2*PI;
+
+			// We don't want to run towards car's direction
+			float dangerZone = angleToFace - vehDirection;
+			dangerZone = CGeneral::LimitRadianAngle(dangerZone);
+
+			// So, add or subtract 90deg (jump to left/right) according to that
+			if (dangerZone <= 0.0)
```
Comment (ShFil119) [inline src/peds/Ped.cpp]: 0.0f

File: src/core/Camera.cpp [hunk 7]
```diff
@@ -1304,6 +1304,17 @@ CCam::GetWeaponFirstPersonOn()
 	return false;
 }
 
+float
+CCamera::Find3rdPersonQuickAimPitch(void)
+{
+	float clampedFrontZ = clamp(Cams[ActiveCam].Front.z, -1.0f, 1.0f);
+
+	// float rot = atan2(clampedFrontZ, sqrt(1.0f - sq(clampedFrontZ)));
+	float rot = asinf(clampedFrontZ);
```
Comment (ShFil119) [inline src/core/Camera.cpp]: Asin

**Comment by erorcun** [comment]
Fixed, thanks.

**Pull request merged**